### PR TITLE
[RHCLOUD-36946] Internal endpoint to update role

### PR DESCRIPTION
**Pull request by lpichler**
## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-36946

## Description of Intent of Change(s)
Update listed attribute(`{"system", "platform_default", "admin_default"}`) for role.

TODO:
- [ ] tests for get and failing cases
 
## Local Testing
How can the feature be exercised?
How can the bug be exploited and fix confirmed?
Is any special local setup required?

## Checklist
- [ ] if API spec changes are required, is the spec updated?
- [ ] are there any pre/post merge actions required? if so, document here.
- [ ] are theses changes covered by unit tests?
- [ ] if warranted, are documentation changes accounted for?
- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?
- [ ] is there known, direct impact to dependent teams/components?
  - [ ] if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by astrozzc — APPROVED** [review]

File: rbac/internal/views.py [hunk 1]
```diff
@@ -697,6 +697,82 @@ def reset_imported_tenants(request: HttpRequest) -> HttpResponse:
     return HttpResponse("Invalid method", status=405)
 
 
+ALLOWED_ROLE_UPDATE_ATTRIBUTES = {"system", "platform_default", "admin_default"}
+
+
+def str_to_bool(value: str) -> bool:
+    """Convert string to bool."""
+    return value.lower() == "true"
```
Comment (petracihalova) [inline rbac/internal/views.py:705]: nitpick: can you trim the empty characters too? `return value.strip().lower() == "true"`

File: tests/internal/test_views.py [hunk 2]
```diff
@@ -1062,3 +1062,117 @@ def test_reset_imported_tenants_excludes_delete(self, delay):
...
+            name="role 1", system=True, tenant=tenant, platform_default=False, admin_default=False
+        )
+
+        request_body = {
+            "system": "false",
+        }
+
+        json_request_body = json.dumps(request_body)
+
+        self.assertEqual(custom_role.system, True)
```
Comment (petracihalova) [inline tests/internal/test_views.py:1079]: nitpick: in this cases you can use `self.assertFalse()` or `self.assertTrue()` too

**Review by petracihalova — APPROVED** [review]
I have only tiny comments + failing linter

**Comment by lpichler** [comment]
@petracihalova Thanks I addressed  your suggestions!

**Review by petracihalova — APPROVED** [review]

**Pull request merged**